image_tools_py/cam2image_py.py

```diff
@@ -0,0 +1,200 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# System imports
+import argparse
+import sys
+
+# Local imports
+import burger_py
+
+# OpenCV imports
+import cv2
+
+# ROS2 imports
+import rclpy
+from rclpy.qos import qos_profile_system_default, qos_profile_sensor_data
+import sensor_msgs.msg
+import std_msgs.msg
+
+
+# Convert an OpenCV matrix encoding type to a string format recognized by
+# sensor_msgs::Image
+def mat2encoding(frame):
+    encoding = ''
+
+    encodings = {1: 'mono', 3: 'bgr', 4: 'rgba'}
+    for channels, prefix in encodings.items():
+        if frame.shape[2] == channels:
+            encoding += prefix
+            break
+    else:
+        raise ValueError('Unsupported frame shape %d' % (frame.shape[2]))
+
+    types = {'uint8': '8', 'int16': '16'}
+    for dtype, num in types.items():
+        if frame.dtype == dtype:
+            encoding += num
+            break
+    else:
+        raise ValueError('Unsupported frame type ' + frame.dtype)
+
+    return encoding
+
+
+# Convert an OpenCV matrix to a ROS Image message.
+def convert_frame_to_message(frame, frame_id, msg):
+    msg.height = frame.shape[0]
+    msg.width = frame.shape[1]
+    msg.encoding = mat2encoding(frame)
+    msg.data = frame.data.tobytes()
+    msg.step = int(len(msg.data) / msg.height)
+    msg.header.frame_id = str(frame_id)
+
+
+def main(args=None):
+    # Pass command-line arguments to rclpy.
+    rclpy.init(args=args)
+
+    # Parse the command-line options.
+    parser = argparse.ArgumentParser(formatter_class=argparse.ArgumentDefaultsHelpFormatter)
+    parser.add_argument(
+        '-b', '--burger', dest='burger_mode', action='store_true', default=False,
+        help='Produce images of burgers rather than connecting to a camera')
+    parser.add_argument(
+        '-d', '--depth', dest='depth', action='store', default=qos_profile_system_default.depth, type=int,
+        help='Queue depth')
+    parser.add_argument(
+        '-f', '--frequency', dest='frequency', action='store', default=30, type=int,
+        help='Publish frequency in Hz')
+    parser.add_argument(
+        '-k', '--keep', dest='history_policy', action='store', default=qos_profile_system_default.history,
+        type=int, choices=[0, 1],
+        help='History QoS setting, 0 - keep last sample, 1 - keep all the samples')
+    parser.add_argument(
+        '-r', '--reliability', dest='reliability_policy', action='store',
+        default=qos_profile_system_default.reliability, type=int, choices=[0, 1],
+        help='Reliability QoS setting, 0 - best effort, 1 - reliable')
+    parser.add_argument(
+        '-s', '--show', dest='show_camera', action='store', default=0, type=int, choices=[0, 1],
```

Comment:
@clalancette why not using `action='store_true'` here instead?

Reply:
Mostly for compatibility with the `image_tools` C++ tools.  There, we parse arguments by hand, and `-s` is an integer argument.  I wanted to make this mirror the C++ tools as much as possible, so while this interface is a-idiomatic, it does make it so you can run the tools interchangeably.

Reply:
Sounds reasonable.